include/gemmi/pdb.hpp

```diff
@@ -283,16 +282,17 @@ Structure read_pdb_from_input(Input&& infile, const std::string& source,
   bool after_ter = false;
   Transform matrix;
   std::unordered_map<ResidueId, int> resmap;
-  while (size_t len = copy_line_from_stream(line, max_line_length+1, infile)) {
+  while (size_t len = copy_line_from_stream(line, max_line_length+1, stream)) {
     ++line_num;
     if (is_record_type(line, "ATOM") || is_record_type(line, "HETATM")) {
-      if (len < 66)
+      if (len < 55)
```

Comment:
Not sure I understand what 66 and 55 represent. Maybe change them to variables (with names)?

Reply:
I think these are the number of bytes (char) in a record (line) of a pdb file that is being parsed. Since it is a fixed file format, I agree with you. This is how I did it in src/core/pdb.cpp (see below) however, this particular file is included from an upstream repository (gemmi project) so I'll leave it alone.

// Define fixed width limits for PDB reading
#define NAMESTART	12  //Char
#define NAMELENGTH	 4
#define RESIDUESTART 16
#define RESIDUELENGTH 5
#define XSTART 		30  // Float
#define YSTART		38
#define ZSTART		46
#define XYZLENGTH	 8
#define OCCUPANCYSTART 54 //Float
#define OCCUPANCYLENGTH 6
#define BFACTORSTART   60 //Float
#define BFACTORLENGTH 	6
#define WATERSTART	   72 // Char, where VMD put WTAA for a TIP3 Water. This is a deprecated segment id spot also used still by Chimera for pdbSegment attribute
#define WATERLENGTH     4
#define ELEMENTSTART   76 // Char, sometimes length = 3, e.g. N1+ or O1- :: I expect not to see O1+ s.t. the charge may be inferred from the extra "1" and element name
#define ELEMENTLENGTH   2
#define CHARGESTART	   78 //Char, not enabled, set to 0.0
#define CHARGELENGTH	2
#define REMARKSTART		7 // Remark ID 351 used to hold particle instances.
#define REMARKLENGTH		4
#define XPARTICLE		11
#define YPARTICLE		19
#define ZPARTICLE		27
#define E1PARTICLE		35
#define E2PARTICLE		43
#define E3PARTICLE		51
#define PARTICLELENGTH	8